optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/ChangeMoveSelectorFactory.java

```diff
@@ -64,6 +66,14 @@ public ChangeMoveSelectorFactory(ChangeMoveSelectorConfig moveSelectorConfig) {
         ValueSelector<Solution_> valueSelector = valueSelectorFactory.buildValueSelector(configPolicy,
                 entitySelector.getEntityDescriptor(),
                 minimumCacheType, SelectionOrder.fromRandomSelectionBoolean(randomSelection));
+        // FIXME ugly
+        if (valueSelector.getVariableDescriptor().isListVariable()) {
+            return new ListChangeMoveSelector<>(
+                    (ListVariableDescriptor<Solution_>) valueSelector.getVariableDescriptor(),
+                    entitySelector,
+                    valueSelector,
+                    randomSelection);
+        }
         return new ChangeMoveSelector<>(entitySelector, valueSelector, randomSelection);
```

Comment:
If it ends up unfolding `@PEntity class Employee{@PCollectionVariable taskList; @PVariable room;}`,
what does it do?
either it behaves correctly (if possible?), or it throws a clean error message? (probably the later)

Reply:
TODO try this manually.

Reply:
Using a `@PlanningCollectionVariable` together with a `@PlanningVariable` on the same entity is currently not supported and fails fast with an explanation.